### PR TITLE
Added looseExact function

### PR DESCRIPTION
**Pull request by beckyconning**

### CODE REVIEW

**Comment by beckyconning** [comment]
@cryogenian please review : )

**Comment by jonsterling** [comment]
@beckyconning Could you clarify the meaning of the term `looseExact`? thanks!

**Comment by beckyconning** [comment]
Yeah I know it's rubbish. My first try was `dnifExact`. I am tired : P.

When you give `findExact` a locator and it can find an element with it then it will give you that element and if it can't find an element it throws an error.

When you give `looseExact` a locator and it can't find an element with it then it will give you unit, if it can find an element it throws an error.

"find" is the opposite of "loose" is the idea. But it its not great : P

**Comment by jonsterling** [comment]
Oh, i see! now that i understand, i actually like the name! But i think it should be spelt "lose" not "loose".

JMS

**Comment by beckyconning** [comment]
Thanks! : )

**Comment by beckyconning** [comment]
I need a function name spellchecker.

**Comment by garyb** [comment]
I think you'd need a "semantic-checker". Not sure they've invented one of those yet :wink:

**Comment by cryogenian** [comment]
Why is it foreign and why do you need it? One can use `findElement` and check for `Nothing` or `findExact` and catch error.

**Comment by cryogenian** [comment]
And by the way here is unsuccessfully named `contra` in `Selenium.Combinators` :smiley:

**Comment by beckyconning** [comment]
Webdriver should throw specific technical errors so we don't have to. Also often you just want to lose an element.

e.g.

```
deleteNote
tryRepeatedlyTo $ byId noteId >>= loseElement
```

**Comment by beckyconning** [comment]
also `contra` eats errors and doesn't have access to the locator.

**Comment by beckyconning** [comment]
Its foreign because of `locator.toString()`. We could add `showLocator` and even a `Show Locator` instance in the future. We should raise a tickets to move a few things away from foreign though.

**Comment by cryogenian** [comment]
`loseElement` doesn't throw any selenium error. 
What do you mean `contra eats error`? 
:+1: for adding `showLocator`

**Comment by beckyconning** [comment]
Contra throws an error like "passed but I am contra so this is not good". Which isn't very informative. 

This is like the purescript webdriver so i consider it a purescript-webdriver error I guess? The implementation is very simple its basically the opposite of findExact with an error that looks a lot like `findExact`'s error.

**Comment by beckyconning** [comment]
Maybe we could merge this for now and raise a ticket to add a `Show Locator` instance and to reimplement `loseElement`'s functionality in Purescript?

**Comment by garyb** [comment]
Probably both of them should share a foreign implementation then, with a function that produces the error.

**Comment by cryogenian** [comment]
I would prefer `loseExact` be written in purescript. But I would also prefer `findElement` be written in purescript. I'll make ticket for it. 
:+1:

**Comment by beckyconning** [comment]
I re-wrote it in purescript : )

File: src/Selenium.js [hunk 1]
```diff
@@ -132,6 +132,12 @@ function _exact(driver) {
     };
 }
 
+exports.showLocator = function(by) {
+    function(cb, eb) {
+        return by.toString();
```
Comment (cryogenian) [inline src/Selenium.js]: you mean `return cb(by.toString())`

Comment (cryogenian) [inline src/Selenium.js]: and it can be just `Locator -> String` I think not `Locator -> Aff _ String`

File: src/Selenium/Monad.purs [hunk 2]
```diff
@@ -218,6 +219,17 @@ saveScreenshot name = getDriver >>= S.saveScreenshot name >>> lift
 findExact :: forall e o. Locator -> Selenium e o Element
 findExact loc = getDriver >>= flip S.findExact loc >>> lift
 
+-- | Tries to find element and throws an error if it succeeds.
+loseElement :: forall e o a. Selenium e o Locator -> Selenium e o Unit
```
Comment (cryogenian) [inline src/Selenium/Monad.purs]: Maybe `Driver -> Locator -> Aff e Unit`?

Comment (cryogenian) [inline src/Selenium/Monad.purs]: and then add it here `getDriver >>= flip S.loseElement >>> lift`

Comment (beckyconning) [inline src/Selenium/Monad.purs]: Driver is in the `Selenium` (`ReaderT`) computation.

Comment (beckyconning) [inline src/Selenium/Monad.purs]: I used `attempt` which needs the `Selenium`. I could do it another way though?

Comment (beckyconning) [inline src/Selenium/Monad.purs]: Or move it to `Combinators`.

**Comment by cryogenian** [comment]
awesome! :+1:

**Pull request merged**